src/coreclr/jit/fgbasic.cpp

```diff
@@ -767,6 +801,39 @@ void Compiler::fgReplacePred(BasicBlock* block, BasicBlock* oldPred, BasicBlock*
     }
 }
 
+//------------------------------------------------------------------------
+// fgReplacePred: redirects the given edge to a new predecessor block
+//
+// Arguments:
+//   edge - the edge whose source block we want to update
+//   newPred - the new predecessor block for edge
+//
+// Notes:
+//
+// This function assumes that all branches from the predecessor (practically, that all
+// switch cases that target the successor block) are changed to branch from the new predecessor,
+// with the same dup count.
+//
+// Note that the successor block's bbRefs is not changed, since it has the same number of
+// references as before, just from a different predecessor block.
+//
+// Also note this may cause sorting of the pred list.
+//
+void Compiler::fgReplacePred(FlowEdge* edge, BasicBlock* const newPred)
+{
+    assert(edge != nullptr);
+    assert(newPred != nullptr);
+    assert(edge->getSourceBlock() != newPred);
+
+    edge->setSourceBlock(newPred);
+
+    // We may now need to reorder the pred list.
+    //
+    BasicBlock* succBlock = edge->getDestinationBlock();
+    assert(succBlock != nullptr);
+    succBlock->ensurePredListOrder(this);
```

Comment:
Speaking of pred list order, we might want to revisit https://github.com/dotnet/runtime/pull/93636 sometime.